### ISSUE TITLE
Fix a bug in Parallelepiped ray_intersection

### ISSUE THREAD

**Issue by angela-bonato** [angela-bonato/RayTracingCourse#9]
While working on v0.3.0, we found a bug in parallelepiped class: ray_intersection and all_ray intersection do not work (see images below). We published v0.2.0 without noticing it so we want to fix this problem, then release v0.2.1 with the correction, then we will release v0.3.0 which is nearly ready a part from this bug.

This was the first time I noticed the bug 
![WhatsApp Image 2024-05-31 at 15 45 28](https://github.com/angela-bonato/RayTracingCourse/assets/129956310/deb13ed4-bc5b-44da-8b2a-d5155d58f75c)

Then I tried to fix it and this is the result....wrong again :(
![mirrors](https://github.com/angela-bonato/RayTracingCourse/assets/129956310/b3ea562e-ac9a-4ac6-91ec-8005a9d17d88)

**Comment by TommasoRovelli** [comment]
Nice shots!

**Comment by angela-bonato** [comment]
Firstly we added some more checks in ray_intersection proc in order to better select the intersections and solve the initial issue. Then linear holes appeared and the problem turned out to be a series of rounding errors. We re-defined <= , >= and == operators and we managed to correct it.

Tadaaa

![mirrors](https://github.com/angela-bonato/RayTracingCourse/assets/129956310/7fa800dc-3d42-4d4e-a5d0-6a280efe7865)

There is a small line in the middle but it is due to the two mirrors intersections, it is not a bug.
